leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -86,6 +89,10 @@ private AwaitingLeadershipProxy(Supplier<T> delegateSupplier,
         this.delegateRef = new AtomicReference<>();
         this.interfaceClass = interfaceClass;
         this.isClosed = false;
+
+        String logTag = "Proxy #" + numProxies.getAndIncrement() + " - " + interfaceClass.getSimpleName();
```

Comment:
this seems like an odd logger name, typically they are dot delimited and I think you'd want the interface class before the counter so that one could adjust log levels for all proxies of specific interface type (e.g. `com.palantir.leader.proxy.AwaitingLeadershipProxy.SomeService.proxy.N`)